source/javascripts/controllers/_LicensesController.js.erb

```diff
@@ -0,0 +1,83 @@
+(function() {
+	"use strict";
+
+	angular
+		.module("BitriseWorkflowEditor")
+		.controller("LicensesController", function(
+			appService,
+			requestService,
+			Workflow,
+			$scope,
+			$rootScope,
+		) {
+			var viewModel = this;
+
+			$scope.licensePoolsUrl = "/workspaces/" + appService.appDetails.ownerData.slug + "/settings/integrations?tab=license-pools";
+
+			$scope.licenses = [
+				{
+					id: null,
+					name: '(No license pool selected)',
+				}
+			];
+
+			requestService.getLicensePools(appService.publicApi, appService.appDetails.ownerData.slug).then(
+				function (response) {
+					var pools = response.licensePools;
+					if (pools.length > 0) {
+						_.each(pools, function(pool) {
+							$scope.licenses.push(pool);
+						});
+					}
+					else {
+						$scope.licenses = [];
+					}
+					$scope.$apply()
```

Comment:
Is this needed? Afaik AngularJS runs a digest loop properly when a service's async request is finished.

Reply:
I don't really understand why, but in some cases, it doesn't work without the manual $apply.  It is easy to reproduce the problem if you take out the manual $apply and try to refresh it a couple of times.